src/aosm/azext_aosm/configuration_models/onboarding_sns_input_config.py

```diff
@@ -42,24 +40,31 @@ def validate(self):
         """Validate the configuration."""
         if not self.location:
             raise ValidationError("Location must be set")
-        if not self.operator_resource_group:
+        if not self.operator_resource_group_name:
             raise ValidationError("Operator resource group name must be set")
         if not self.site_name:
             raise ValidationError(
                 "site name must be set"
             )
+        if not self.nsd_reference:
+            raise ValidationError("At least one Network Service Design Version must be included.")
+        self.nsd_reference.validate()
 
+    def __post_init__(self):
+        if self.nsd_reference is None:
```

Comment:
I think this needs to be a loop, as you can have multiple nsds references right? See the CNF for that

Reply:
only one NSD